File: competitive_programming/leetcode/07_typescript/217_contains_duplicate.ts
function containsDuplicate(nums: number[]): boolean {
  return new Set(nums).size !== nums.length;
}

if (containsDuplicate([1, 2, 3, 1]) !== true || containsDuplicate([1, 2, 3, 4]) !== false) throw new Error("contains duplicate failed");
console.log("217 contains duplicate ok");

File: competitive_programming/leetcode/03_cpp/217_contains_duplicate.cpp
#include <algorithm>
#include <cassert>
#include <cmath>
#include <cstdint>
#include <iostream>
#include <set>
#include <string>
#include <vector>

bool containsDuplicate(const std::vector<int> &nums) {
    std::set<int> seen;
    for (int value : nums) { if (!seen.insert(value).second) return true; }
    return false;
}

int main() {
    assert(containsDuplicate({1, 2, 3, 1}) && !containsDuplicate({1, 2, 3, 4}));
    std::cout << "217 contains duplicate ok\n";
    return 0;
}
